changedetectionio/tests/test_xpath_selector.py

```diff
@@ -277,3 +277,62 @@ def test_check_with_prefix_include_filters(client, live_server):
     assert b"Some text that will change" not in res.data #not in selector
 
     client.get(url_for("form_delete", uuid="all"), follow_redirects=True)
+
+def test_allow_server_respond_with_bytes(client, live_server):
+    res = client.get(url_for("form_delete", uuid="all"), follow_redirects=True)
+    assert b'Deleted' in res.data
+
+    # Give the endpoint time to spin up
+    time.sleep(1)
+
+    # A poorly configured non-UTF-8 HTML of server-side.
+    KR_STR = '혼돈은 당연하다'
+    KR_BYTES_EUC_KR = KR_STR.encode('euc-kr')
+    d = b'''<html lang="ko">
+<head>
+<meta http-equiv="Content-Type" content="text/html; charset=EUC-KR">
+<style>
+p {
+  @charset EUC-KR;
+  color: orange;
+  }
+</style>
+</head>
+<body>
+<p>%b..</p>
+<p>If you are reading this, then server sent bytes successfully.</p>
+</body>
+</html>
+''' % KR_BYTES_EUC_KR
+
+    # This file cannot be read with UTF-8.
+    with open("test-datastore/endpoint-content.txt", "wb") as f:
+        f.write(d)
+
+    # Add our URL to the import page
+    test_url = url_for('test_endpoint', _external=True)
+    res = client.post(
+        url_for("import_page"),
+        data={"urls": test_url},
+        follow_redirects=True
+    )
+    assert b"1 Imported" in res.data
+    time.sleep(3)
+
+    res = client.post(
+        url_for("edit_page", uuid="first"),
+        data={"include_filters":  "xpath://p", "url": test_url, "tags": "", "headers": "", 'fetch_backend': "html_requests"},
+        follow_redirects=True
+    )
+
+    assert b"Updated watch." in res.data
+    time.sleep(3)
+
+    res = client.get(
+        url_for("preview_page", uuid="first"),
+        follow_redirects=True
+    )
+
+    assert b'If you are reading this, then server sent bytes successfully.' in res.data #in selector
```

Comment:
The reason I didn't test 'assert KR_STR.encode() in res.data #in selector' is
    because this program doesn't parse bytes data currently.
    [the commit](https://github.com/dgtlmoon/changedetection.io/pull/1835)
    will change it.